types/yup/index.d.ts

```diff
@@ -105,39 +107,49 @@ export interface StringSchemaConstructor {
     new (): StringSchema;
 }
 
-export interface StringSchema extends Schema<string> {
-    length(limit: number | Ref, message?: TestOptionsMessage): StringSchema;
-    min(limit: number | Ref, message?: TestOptionsMessage): StringSchema;
-    max(limit: number | Ref, message?: TestOptionsMessage): StringSchema;
+export interface StringSchema<T = string> extends Schema<T> {
```

Comment:
I think, here, instead of defaulting it to string, we should say that `T extends string` because we're always going to want the user of `StringSchema` to pass us something that is derived from string specifically.

Reply:
@YashdalfTheGray Looks like I need to change the definition to `interface StringSchema<T extends string | null = string> extends Schema<T>` because it can be null as well as a string type. And defaults to string is none specified.

Reply:
I'm okay with this. What do y'all think? I like that we are forcing the client of this library to pass in something that has known string methods so that the validator will work properly but it might be too restrictive possibly? Thoughts?

Reply:
I think its fair to enforce string type here.
I didn't think of this before, but does 
`interface StringSchema<T extends string | null = string> extends Schema<T>`
prevent me from passing undefined into the schema? if so, should it be 
`interface StringSchema<T extends string | null | undefined = string> extends Schema<T>` instead?

Reply:
@sseppola Yes it should prevent that. When using `--strictNullChecks` both `null` and `undefined` are distinctly different types. But that said when would you pass undefined into the `StringSchema` interface? As far as I can tell that should never happen at the moment. It will when I do the second PR with `notRequired()`/`required()` but I will do that as a separate PR.

Reply:
I tried to make a codesandbox to explore the default behavior of Yup:
https://codesandbox.io/s/formik-yup-form-dskdh

In summary
````
import * as Yup from "yup";

async function validate(value) {
  try {
    const res = await Yup.string().isValid(value);
    console.log(`"${value}" is valid`);
  } catch (error) {
    console.log(`"${value}" is not valid`);
  }
}
validate(undefined);
validate(null);
validate("");

```
the output is:
```
"undefined" is valid
"null" is not valid
"" is valid
```

So to be in line with Yup the type should be even more complicated:
`interface StringSchema<T extends string | null | undefined = string | undefined> extends Schema<T>`

However, as we go from `interface StringSchema extends Schema<string>`, which enforces string, then what you have currently is fine, and we can deal with `undefined` in another PR :)